tools/tool_recommendation_model/create_tool_recommendation_model.xml

```diff
@@ -0,0 +1,111 @@
+
+            <param name="learning_rate" type="text" value="0.0001,0.1" label="Learning rate" help="Provide a range of positive real numbers to sample the learning rate. Learning rate defines the speed of neural network learning. A higher value will ensure fast learning and smaller value will ensure slower learning. An example: 0.0001,0.1"/>
+            
+            <param name="activation_recurrent" type="text" value="elu" label="Name of the activation function for recurrent layers" help="It is a mathematical function that transforms the input of recurrent layers to the following neural network layer."/>
+            <param name="activation_output" type="text" value="sigmoid" label="Name of the activation function for output layer" help="It is a mathematical function that transforms the input of the last dense layer to the output of the neural network."/>
+            <param name="loss_type" type="text" value="binary_crossentropy" label="Name of the loss function" help="The loss/error function computes an error between the true and predicted output. This error is minimised during neural network learning to be as close to the true output as possible. Root Mean Square Propagation (RMSProp) is used as an optimiser to minimise error computed by this error function."/>
+        </section>
+    </inputs>
+    <outputs>
+        <data format="h5" name="outfile_model" label="Model to recommend tools in Galaxy"></data>
+    </outputs>
+    <tests>
+        <test>
+            <param name="input_tabular_workflows" value="test_workflows" ftype="tabular"/>
+            <param name="input_tabular_tool_usage" value="test_tool_usage" ftype="tabular"/>
+            <output name="outfile_model">
+                <assert_contents>
+                    <has_h5_keys keys="best_parameters,class_weights,compatible_tools,data_dictionary,model_config,weight_0,weight_1,weight_2,weight_3,weight_4,weight_5,weight_6,weight_7,weight_8"/>
+                </assert_contents>
+            </output>
+        </test>
+    </tests>
+    <help>
+        <![CDATA[
+**What it does**
+It creates a model to recommend tools in Galaxy by learning the connections of tools in workflows. The model is an HDF5 file containing the tool dictionary, weights and configuration of the neural network. The recurrent neural network (Gated Recurrent Units) is used as a deep learner to learn the higher-order dependencies in tool connections of workflows. It takes two tabular files as input - one for the workflows and another for tools' usage frequencies. There are multiple other parameters to be set to find the best configuration of parameters of the neural network.
```

Comment:
It would be useful if you could add a detailed description of how someone could get the needed input files ...

E.g. this tool assumes that you have access to the DB of your own Galaxy server and can run the tool under /scripts/.... to extract the data .... blable ... the two files that are needed are then produced and should look like ....

File one is .... file two is ....